src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -68,6 +68,41 @@ public function __construct($class = null, array $arguments = [])
         $this->arguments = $arguments;
     }
 
+    public function __sleep()
+    {
+        if ($this->hasErrors()) {
+            //By calling it, we resolve all potentials \Closure inside $this->>errors
+            $this->getErrors();
+        }
+
+        return [
```

Comment:
array_keys(get_object_vars())?

Reply:
yes I've tested that but look à my comment under